tests/test_ote_training.py

```diff
@@ -0,0 +1,613 @@
+
+        self.template = None
+        self.environment = None
+        self.task = None
+        self.output_model = None
+        self.evaluation_performance = None
+        self.environment_for_export = None
+        self.exported_model = None
+        self.openvino_task = None
+        self.evaluation_performance_exported = None
+
+        self.was_training_run = False
+        self.stored_exception_training = None
+        self.was_export_run = False
+        self.stored_exception_export = None
+
+        self.copy_hyperparams = None
+
+    @staticmethod
+    def _create_environment_and_task(params, labels_schema, model_template):
+        environment = TaskEnvironment(model=None, hyper_parameters=params, label_schema=labels_schema,
+                                      model_template=model_template)
+        logger.info('Create base Task')
+        task_impl_path = model_template.entrypoints.base
+        task_cls = get_task_class(task_impl_path)
+        task = task_cls(task_environment=environment)
+        return environment, task
+
+    def _run_ote_training(self):
+        logger.debug(f'self.template_file_path = {self.template_file_path}')
+        logger.debug(f'Using for train annotation file {self.dataset_params.annotations_train}')
+        logger.debug(f'Using for val annotation file {self.dataset_params.annotations_val}')
+
+        self.dataset = MMDatasetAdapter(
+            train_ann_file=self.dataset_params.annotations_train,
+            train_data_root=self.dataset_params.images_train_dir,
+            val_ann_file=self.dataset_params.annotations_val,
+            val_data_root=self.dataset_params.images_val_dir,
+            test_ann_file=self.dataset_params.annotations_test,
+            test_data_root=self.dataset_params.images_test_dir,
+            dataset_storage=NullDatasetStorage)
+
+        self.labels_schema = generate_label_schema(self.dataset.get_labels())
+        labels_list = self.labels_schema.get_labels(False)
+        self.dataset.set_project_labels(labels_list)
+        print(f'train dataset: {len(self.dataset.get_subset(Subset.TRAINING))} items')
+        print(f'validation dataset: {len(self.dataset.get_subset(Subset.VALIDATION))} items')
+
+        logger.debug('Load model template')
+        self.model_template = parse_model_template(self.template_file_path)
+
+        hyper_parameters = self.model_template.hyper_parameters.data
+        set_values_as_default(hyper_parameters)
+
+        logger.debug('Setup environment')
+        params = create(hyper_parameters)
+        logger.debug('Set hyperparameters')
+        params.learning_parameters.num_iters = self.num_training_iters
+        if self.num_training_iters < 20:
+            num_checkpoints = 2
+        elif self.num_training_iters < 1000:
+            num_checkpoints = 10
+        else:
+            num_checkpoints = 30
+
+        #### TODO: fixed parameter, delete this
+        params.learning_parameters.batch_size = 2
```

Comment:
Will you remove this?

Reply:
I will remove this after refactoring. It will be the next PR.